File: js/task2/S2_FOR_04.js
// S2_FOR_04 — Count occurrences

function countOccurrences(values) {
  const counts = {};

  for (const v of values) {
    if (counts[v] === undefined) {
      counts[v] = 1;
    } else {
      counts[v]++;
    }
  }

  return counts;
}

// Tests
console.log(countOccurrences([1, 2, 1, 3, 2, 1]));
// { '1': 3, '2': 2, '3': 1 }

console.log(countOccurrences(["a", "b", "a", "c", "b"]));
// { a: 2, b: 2, c: 1 }

console.log(countOccurrences([]));
// {}

File: python/task2b/S2_FOR_04.py
# S2_FOR_04 — Count occurrences (histogram)

def count_occurrences(values):
    counts = {}
    for v in values:
        if v in counts:
            counts[v] += 1
        else:
            counts[v] = 1
    return counts


def main():
    tests = [
        [1, 2, 2, 3, 1, 2],
        ["a", "b", "a", "c", "b", "a"],
        [],
        [True, False, True, True],
    ]

    for t in tests:
        print(f"in={t} -> counts={count_occurrences(t)}")


if __name__ == "__main__":
    main()
